common/execution_context.go

```diff
@@ -289,34 +288,30 @@ func (e *ExecutionContext) getInjectedScript(apiCtx context.Context) (JSHandleAP
 
 // Eval evaluates the provided JavaScript within this execution context and
 // returns a value or handle.
-func (e *ExecutionContext) Eval(
-	apiCtx context.Context, js goja.Value, args ...goja.Value,
-) (any, error) {
+func (e *ExecutionContext) Eval(apiCtx context.Context, js string, args ...any) (any, error) {
 	opts := evalOptions{
 		forceCallable: true,
 		returnByValue: true,
 	}
 	evalArgs := make([]any, 0, len(args))
 	for _, a := range args {
-		evalArgs = append(evalArgs, a.Export())
+		evalArgs = append(evalArgs, a)
```

Comment:
By assertion, do you mean `panic`?

Reply:
Here's an error after adding assertions.

```
--- FAIL: TestLocator (27.91s)
    --- FAIL: TestLocator/DispatchEvent (0.99s)
dispatching event "click" to "#link": converting argument "mouseevent" in execution context ID 3 and frame ID D3E4FB73C8754085E9ECF21400A060E4: goja.Value escaped [recovered]
```

From here:

```
"DispatchEvent", func(tb *testBrowser, p *common.Page) {
	...
	p.Locator("#link", nil).DispatchEvent("click", tb.toGojaValue("mouseevent"), nil)
	...
	},
```

It might make sense convert these rest of the methods to goja-free at this point.

Reply:
We'll tackle this in #1182. The code in this PR works, even if it receives Goja values from the other parts of the module. And it should be safe because we use the CDP runtime constructs in `Eval` and converters.

Reply:
See #1191.